### ISSUE TITLE
Makefile bugs

### ISSUE THREAD

**Issue by shane-mckinney** [lanl/FEHM#21]
Debug option compiles .o's in -O2 mode

It would be preferable to be able to compile into a build/ dir

**Comment by daniellivingston** [comment]
Should we only have the executable put into `build/`, or should `*.o` files go there as well? I don't think there are any other build artifacts.

**Comment by daniellivingston** [comment]
Note that if we just want the binary only in `build/`, then something like this could be done:

```bash
$ mkdir build/
$ cd src && make all EXE=../build/xfehm_v3.3.1
```

**Comment by shane-mckinney** [comment]
why would you just want the binary in there? build directories hold all the built things! i think its nice for git tracking because you can untrack all of the build related files easier? other thansomething like that its nice to have auto complete not get stuck on *.f vs *.o , also if youre looking for a file name in there its already messy enough as is without all the the .mod's and .o's


Shane McKinney
Los Alamos National Laboratory
EES-16
Computational Earth Sciences
smckinney@lanl.gov
292026
505-412-9844
________________________________

**Comment by daniellivingston** [comment]
> i think its nice for git tracking because you can untrack all of the build related files easier

The `.gitignore` file is configured to not track build artifacts:

```
*.o
*.mod
debug/*
build/*
src/dated.f
*.pyc
*DS_Store
```

But yeah, valid point. @millerta , thoughts building the binary and build artifacts into `build/` instead of `src/`?

**Comment by daniellivingston** [comment]
@shane-mckinney The `make debug` problem is fixed as of 6fa70575faeca51c62ce4047409aba6d85c0452f. Take a pull and see if this is what you're after.
